soccer/src/soccer/strategy/agent/position/robot_factory_position.hpp

```diff
@@ -110,16 +110,12 @@ class RobotFactoryPosition : public Position {
         current_position_->send_pass_confirmation(target_robot);
     }
 
+    void set_override_position(const strategy::OverridingPositions overriding_position) override;
+
 private:
     std::unique_ptr<Position> current_position_;
```

Comment:
```suggestion
    std::unique_ptr<RobotFactoryPosition> current_position_;
```
this would literally be enough for now. but taking out the pointer also possible. definitely not a fan of the function being in Position when its clearly not a generic Position task